src/components/BulkOperationDetailsPage/BulkOperationDetails.jsx

```diff
@@ -0,0 +1,155 @@
+import React, { useState } from 'react';
+import 'moment-timezone';
+import PropTypes from 'prop-types';
+import {
+  Button, DataTable, useToggle, FullscreenModal, Stack,
+} from '@openedx/paragon';
+
+import { formatDateTime } from '../../utils';
+import './styles.scss';
+
+const ImageCell = ({ value }) => (
+  value ? (
+    <a href={value} target="_blank" rel="noopener noreferrer">
+      View Image
+    </a>
+  ) : null
+);
+
+ImageCell.propTypes = {
+  value: PropTypes.string,
+};
+
+ImageCell.defaultProps = {
+  value: null,
+};
+
+const parseCSV = async (url) => {
+  const response = await fetch(url);
+  const text = await response.text();
+  const lines = text.trim().split('\n');
+
+  const headers = lines[0].split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+  const cleanedHeaders = headers.map((h) => h.replace(/(^"|"$)/g, ''));
+
+  const rows = lines.slice(1).map((line) => line.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/).map((cell) => cell.replace(/(^"|"$)/g, '')));
+
+  return { headers: cleanedHeaders, rows };
+};
+
+const BulkOperationDetails = ({ task }) => {
+  const [csvRows, setCsvRows] = useState([]);
+  const [csvHeaders, setCsvHeaders] = useState([]);
+  const [isOpen, open, close] = useToggle(false);
+  const DEFAULT_CSV_PREVIEW_PAGE_SIZE = 5;
+
+  if (!task || Object.keys(task).length === 0) {
+    return <p className="text-info"> No BulkOperationTask found against the given ID. Please check the ID and try again.</p>;
+  }
+
+  const handlePreviewCSV = async () => {
+    try {
+      const { headers, rows } = await parseCSV(task.csv_file);
+      setCsvHeaders(headers);
+      setCsvRows(rows);
+    } catch (error) {
+      console.error('Failed to preview CSV:', error);
+    }
+  };
+
+  const getDataTableColumns = (headers) => headers.map((header) => {
+    const isImageColumn = header.toLowerCase() === 'image';
+
+    const column = {
+      Header: header,
+      accessor: header,
+    };
+
+    if (isImageColumn) {
+      column.Cell = ImageCell;
+    }
+
+    return column;
+  });
+
+  return (
+    <div className="container mt-2">
+      <h2 className="mb-4">Task Details</h2>
+      <p><strong>Task ID:</strong> {task.task_id}</p>
+      <p><strong>Task type:</strong> {task.task_type}</p>
+      <p><strong>Uploaded by:</strong> {task.uploaded_by}</p>
+      <p><strong>Status:</strong> {task.status}</p>
+      <p><strong>Created:</strong> {task && formatDateTime(task.created)}</p>
+      <p><strong>Modified:</strong> {task && formatDateTime(task.modified)}</p>
```

Comment:
nit: you can use table instead.

Reply:
I like the current imp better than a table. However, I'll update the code and add a screenshot here, then we can decide.

Reply:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/74142831-9b90-4919-85f1-e1c470e37fb8" />

Reply:
Well, this looks better.